torch/signal/windows/windows.py

```diff
@@ -296,3 +297,84 @@ def gaussian(
                        requires_grad=requires_grad)
 
     return torch.exp(-k ** 2)
+
+
+@_add_docstr(
+    r"""
+Computes the Kaiser window.
+
+The Kaiser window is defined as follows:
+
+.. math::
+    out_i = I_0 \left( \beta \sqrt{1 - \left( {\frac{i - N/2}{N/2}} \right) ^2 } \right) / I_0( \beta )
+
+where ``I_0`` is the zeroth order modified Bessel function of the first kind (see :func:`torch.special.i0`), and
+``N = M - 1 if sym else M``.
+
+``M`` is the window length.
+
+    """,
+    r"""
+
+{normalization}
+
+Args:
+    {M}
+
+Keyword args:
+    beta (float, optional): shape parameter for the window. Default: 12.0
+    {sym}
+    {dtype}
+    {layout}
+    {device}
+    {requires_grad}
+
+Examples::
+
+    >>> # Generate a symmetric gaussian window with a standard deviation of 1.0.
+    >>> torch.signal.windows.kaiser(5)
+    tensor([4.0065e-05, 2.1875e-03, 4.3937e-02, 3.2465e-01, 8.8250e-01, 8.8250e-01, 3.2465e-01, 4.3937e-02, 2.1875e-03, 4.0065e-05])
+
+    >>> # Generate a periodic gaussian window and standard deviation equal to 0.9.
+    >>> torch.signal.windows.kaiser(5,sym=False,std=0.9)
+    tensor([1.9858e-07, 5.1365e-05, 3.8659e-03, 8.4658e-02, 5.3941e-01, 1.0000e+00, 5.3941e-01, 8.4658e-02, 3.8659e-03, 5.1365e-05])
+""".format(
+        **window_common_args,
+    ),
+)
+def kaiser(
+        M: int,
+        *,
+        beta: float = 12.0,
+        sym: bool = True,
+        dtype: Optional[torch.dtype] = None,
+        layout: torch.layout = torch.strided,
+        device: Optional[torch.device] = None,
+        requires_grad: bool = False
+) -> Tensor:
+    if dtype is None:
+        dtype = torch.get_default_dtype()
+
+    _window_function_checks('kaiser', M, dtype, layout)
+
+    if M == 0:
+        return torch.empty((0,), dtype=dtype, layout=layout, device=device, requires_grad=requires_grad)
+
+    if M == 1:
+        return torch.ones((1,), dtype=dtype, layout=layout, device=device, requires_grad=requires_grad)
+
+    if beta <= 0:
```

Comment:
check for beta should go before returning the wrong value for M=1

Reply:
What's interesting is that SciPy does not require a positive beta:

```
scipy.signal.windows.kaiser(1, beta=-5)
: array([1.])

scipy.signal.windows.kaiser(10, beta=-5.)
: array([0.03671089, 0.20127873, 0.47552746, 0.7753221 , 0.97273069,
       0.97273069, 0.7753221 , 0.47552746, 0.20127873, 0.03671089])
```

The wikipedia entry stipulates that beta should be non-negative, but not strictly positive: https://en.wikipedia.org/wiki/Kaiser_window

So, follow-up questions:

-  why are we requiring a positive beta?
- should the documentation discuss this requirement?

Reply:
Sure, let's make it non-negative. Definition includes I0(beta) and I0 is not defined for negative args (and there's no alternative definition for 1), so looks like scipy is wrong.

Reply:
Agreed with @ngimel. I'll turn that precheck into a non-negative one.

Reply:
@lezcano @mruberry @ngimel I guess for the gaussian and exponential windows we should move the `std` and `tau` check up a few lines too.

Reply:
yup. Feel free to add it to this PR if oyu want.

Reply:
I'll do it in #87082

Reply:
> Agreed with @ngimel. I'll turn that precheck into a non-negative one, and I'll move it up as well.

But why not just allow negative betas, too?

Reply:
@mruberry 

![image](https://user-images.githubusercontent.com/13088001/198110555-65dbeaa0-b8f1-4516-8c98-a1056e138bdb.png)

Reply:
I understand, just interesting that SciPy allows them